hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkGetAccountDetailsHandler.java

```diff
@@ -16,54 +16,271 @@
+import static com.hedera.node.app.spi.key.KeyUtils.isEmpty;
+import static com.hedera.node.app.spi.validation.Validations.mustExist;
 import static java.util.Objects.requireNonNull;
 
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.Duration;
 import com.hedera.hapi.node.base.HederaFunctionality;
 import com.hedera.hapi.node.base.QueryHeader;
 import com.hedera.hapi.node.base.ResponseHeader;
+import com.hedera.hapi.node.base.ResponseType;
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.hapi.node.base.TokenFreezeStatus;
+import com.hedera.hapi.node.base.TokenID;
+import com.hedera.hapi.node.base.TokenKycStatus;
+import com.hedera.hapi.node.base.TokenRelationship;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.hapi.node.state.token.TokenRelation;
+import com.hedera.hapi.node.token.AccountDetails;
+import com.hedera.hapi.node.token.GetAccountDetailsQuery;
 import com.hedera.hapi.node.token.GetAccountDetailsResponse;
+import com.hedera.hapi.node.token.GrantedCryptoAllowance;
+import com.hedera.hapi.node.token.GrantedNftAllowance;
+import com.hedera.hapi.node.token.GrantedTokenAllowance;
 import com.hedera.hapi.node.transaction.Query;
 import com.hedera.hapi.node.transaction.Response;
+import com.hedera.node.app.service.evm.contracts.execution.StaticProperties;
+import com.hedera.node.app.service.networkadmin.impl.config.NetworkAdminServiceConfig;
+import com.hedera.node.app.service.networkadmin.impl.utils.NetworkAdminServiceUtil;
+import com.hedera.node.app.service.token.ReadableAccountStore;
+import com.hedera.node.app.service.token.ReadableTokenRelationStore;
+import com.hedera.node.app.service.token.ReadableTokenStore;
+import com.hedera.node.app.service.token.ReadableTokenStore.TokenMetadata;
 import com.hedera.node.app.spi.workflows.PaidQueryHandler;
 import com.hedera.node.app.spi.workflows.PreCheckException;
 import com.hedera.node.app.spi.workflows.QueryContext;
+import com.hedera.node.config.data.LedgerConfig;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
 import javax.inject.Inject;
 import javax.inject.Singleton;
 
 /**
- * This class contains all workflow-related functionality regarding {@link HederaFunctionality#GET_ACCOUNT_DETAILS}.
+ * This class contains all query information regarding {@link HederaFunctionality#GET_ACCOUNT_DETAILS}.
  */
 @Singleton
 public class NetworkGetAccountDetailsHandler extends PaidQueryHandler {
+
     @Inject
     public NetworkGetAccountDetailsHandler() {
-        // Exists for injection
+        // exists for injection
     }
 
     @Override
+    @NonNull
     public QueryHeader extractHeader(@NonNull final Query query) {
         requireNonNull(query);
-        return query.accountDetailsOrThrow().header();
+        return requireNonNull(query.accountDetailsOrThrow().header());
```

Comment:
Instead of `requireNonNull`, you could just call `headerOrThrow`...

Reply:
fixed it thank you